astropy/io/ascii/ecsv.py

```diff
@@ -131,6 +156,9 @@ def get_cols(self, lines):
             raise core.InconsistentTableError(no_header_msg)
         # ecsv_version could be constructed here, but it is not currently used.
 
+        self.ecsv_version = tuple(v for v in match.groups() if v is not None)
```

Comment:
This normalizes the version to a 3-tuple of ints.
```suggestion
        self.ecsv_version = tuple(int(v or 0) for v in match.groups())
```